modules/listener/event_types.bal

```diff
@@ -0,0 +1,197 @@
+    string token;
+    string challenge;
+    string 'type;
+|};
+
+public type AppEvent record {
+    string 'type;
+    string user;
+    string channel;
+    string event_ts;
+    string tab?;
+    string text?;
+    string ts?;
+};
+
+public type CallEvent record {
+    string 'type;
+    string call_id;
+    string user_id;
+    string channel_id;
+    string external_unique_id;
+};
+
+public type MessageEvent record {
+    string 'type;
+    string channel;
+    string user;
+    string text;
+    string ts;
+    Edited edited?;
+    string subtype?;
+    boolean hidden?;
+    string deleted_ts?;
+    string event_ts?;
+    boolean is_starred?;
+    string channel_type?;
+};
+
+public type MemberEvent record {
+    string 'type;
+    string user;
+    string channel;
+    string channel_type?;
+    string team?;
+    string inviter?;
+};
+
+public type DNDEvent record {
+    string 'type;
+    string user;
+    DND_Detail dnd_status;
+};
+
+public type GridMigrationEvent record {
+    string 'type;
+    string enterprise_id;
+};
+
+public type InviteRequestedEvent record {
+    string 'type;
+    InviteRequest invite_request;
+};
+
+public type ReactionEvent record {
+    string 'type;
+    string user;
+    string reaction;
+    string item_user?;
+    ReactionItem item;
+    string event_ts;
+};
+
+public type FileEvent record {
+    string 'type;
+    string file_id;
+    FileDetails file?;
+    string|string[]|json|json[] comment?;
+    string event_ts?;
+};
+
+public type GenericSlackEvent record {
+    string 'type;
+    string|Channel channel?;
+    string user?;
+    string latest?;
+    string ts?;
+    string event_ts?;
+    string connected_team_id?;
+    string previously_connected_team_id?;
+    boolean is_ext_shared?;
+    string subtype?;
+    string[] names?;
+    string message_ts?;
+    string thread_ts?;
+    json[]|json links?;
+    boolean is_bot_user_member?;
+    boolean has_pins?;
+    string|string[] scopes?;
+    string trigger_id?;
+    json[]|json item?;
+    string subteam_id?;
+};
+
+public type Edited record {
+    string user;
+    string ts;
+};
+
+public type Channel record {
+    string id;
+    string name;
+    int created;
+    string creator?;
+};
+
+public type DND_Detail record {
+    boolean dnd_enabled;
+    int? next_dnd_start_ts;
+    int? next_dnd_end_ts;
+    boolean snooze_enabled?;
+    int? snooze_endtime?;
+};
+
+public type InviteRequest record {
+    string id;
+    string email;
+    int date_created;
+    string[] requester_ids;
+    string[] channel_ids;
+    string invite_type;
+    string real_name;
+    int date_expire;
+    string request_reason?;
+    Team team?;
+};
+
+public type Team record {
+    string id;
+    string name?;
+    string domain?;
+};
+
+public type FileDetails record {
+    string id;
+};
+
+public type ReactionItem record {
+    string 'type;
+    string channel?;
+    string ts?;
+    string file?;
+    string file_comment?;
+};
+
```

Comment:
**Please review here**
Here slack event types are categorized according to the common keys identified.

Reply:
Can't we catagorize using map ? . Suggestion:
```
map<string[]> eventTypes = {
    TYPE = [APPEVENT, ...REACTIONEVENT],
    USER = [APPEVENT, GENERICSLACKEVENT....],
   .....
  };
//need to set constants for the keys and the eventName, We can have full control and easy to maintinan
```

Reply:
The main reason for categorizing event type names is to treat a category as a type to check in a 'if statement' as following
 ```if (eventType is MessageEventType)```

If we use a map to categorize these name we need to use it in our 'if statement' as following. 
```
if(lang.array:indexOf(eventTypes["TYPE"], eventType) != ())
```
The current way seems simpler. Please convey your opinion.